File: Behavioral/CommandPattern(Behavioral)/CommandPatternWithHistory/undo_redo_java/PopCommand.java
import java.util.Stack;

public class PopCommand implements Command {
    private final Stack<Integer> stack;
    private Integer i = null;

    public PopCommand(Stack<Integer> stack) {
        this.stack = stack;
    }

    @Override
    public void execute() {
        if (!stack.isEmpty()) {
            i = stack.pop();
        }
    }

    @Override
    public void unExecute() {
        if (i != null) {
            stack.push(i);
        }
    }
}

File: Behavioral/CommandPattern(Behavioral)/CommandPatternWithHistory/undo_redo/PopCommand.ts
import { Command } from "./Command";

export class PopCommand implements Command {
  private i: number | undefined;
  constructor(private stack: number[]) {}

  execute(): void {
    this.i = this.stack.pop();
  }

  unExecute(): void {
    if (this.i !== undefined) {
      this.stack.push(this.i);
    }
  }
}
